server/plugin/api.go

```diff
@@ -626,42 +655,48 @@ func (p *Plugin) handleFileUpload(c *Context, w http.ResponseWriter, r *http.Req
 	err := json.NewDecoder(r.Body).Decode(&request)
 	if err != nil {
 		p.API.LogError("Failed to decode SubmitDialogRequest", "err", err)
-		w.WriteHeader(http.StatusBadRequest)
+		p.writeInteractiveDialogError(w, DialogErrorResponse{Error: "Failed to decode request", StatusCode: http.StatusBadRequest})
 		return
 	}
 	defer r.Body.Close()
 
 	fileID := request.Submission["fileID"].(string)
 	fileInfo, appErr := p.API.GetFileInfo(fileID)
 	if appErr != nil {
-		p.API.LogError("unable to fetch file info", "err", err, "fileID", fileID)
-		w.WriteHeader(http.StatusInternalServerError)
+		p.API.LogError("Unable to fetch file info", "err", appErr, "fileID", fileID)
+		p.writeInteractiveDialogError(w, DialogErrorResponse{Error: "Unable to fetch file info", StatusCode: http.StatusInternalServerError})
```

Comment:
Unsure if we should be this explicit with errors for end users or if we should just put that something went wrong and that they should contact and administrator for assistance. The admins would have the error logs for debugging. 3/5

Reply:
Yeah you're correct, I'll adjust those

Reply:
I adjusted all of those to be the same error message